core/arch/arm/plat-imx/registers/imx6.h

```diff
@@ -140,4 +140,29 @@
 #define DIGPROG_OFFSET	0x260
 #endif
 
+#if defined(CFG_MX6ULL)
+#define I2C1_BASE		0x021a0000
```

Comment:
sure, that can be done.
why do you say that CFG_IMX_I2C_CLK_RATE is the same for all supported platforms? where did you get that information from? the peripheral clock rate is not always configured  with the oscillator clock (certainly not in mainline u-boot)

Reply:
yes that is the idea: to be able to work around bootloaders configuring the peripheral clock at a different rate (nowhere says that 24MHz is mandatory).
so where do you want this clock config?

Reply:
Well, I would leave the default clock speed as a ```#define ...``` like it was before.
I would like to avoid defining the clock speed of a peripheral in ```conf.mk``` becoming the norm ... I don't know where we should move this.
Is the default clock speed working with se050?
If you need to modify the clock speed, is it possible to keep the modification in your own tree?

Reply:
> I would like to avoid defining the clock speed of a peripheral in `conf.mk` becoming the norm ...

What's wrong with that?

Reply:
There are already a lot of stuff in ```conf.mk```
If we go in this direction, we could define a default clock speed for each peripheral and/or each platform ... adding even more things to this file. That's something I would like to avoid if possible.

Reply:
> There are already a lot of stuff in `conf.mk`

Well, if it is not in `conf.mk` it will be somewhere else ;-) If things get too big we can always create separate `.mk` files and include them (one `.mk` file for a specific device/driver for example).

> If we go in this direction, we could define a default clock speed for each peripheral and/or each platform 

;-) depends on the device I suppose.
The main thing I would like to emphasise is: in general, fixed parameters are best defined in header files (with `#define`), but configurable parameters are much easier in `.mk` files because with one line `CFG_SOMETHING ?= value` you can define a default value, let the user override it easily if needed, and make it available to makefiles and C code.

That being said I have no strong opinion, and @clementfaure I will happily leave you the final word.